pytrickle/health.py

```diff
@@ -0,0 +1,122 @@
+"""
+Health management for trickle streaming applications.
+
+This module provides a reusable health state manager that can track the health
+of streaming applications including pipeline status, active streams, and errors.
+"""
+
+import logging
+from typing import Optional, Dict, Any
+
+logger = logging.getLogger(__name__)
+
+class StreamHealthManager:
+    """Manages the health state of a streaming application."""
+    
+    def __init__(self, service_name: str = "trickle-stream-service"):
+        self.service_name = service_name
+        self.state = "LOADING"  # Initial state during startup
+        self.error_message = None
+        self.is_pipeline_warming = False
+        self.pipeline_ready = False
+        self.active_streams = 0
```

Comment:
Not sure we need to track active streams here, and looks like the application has to track it anyways.

Reply:
`StreamHealthManager` is updated to use the new `PipelineState` enum 
https://github.com/livepeer/pytrickle/blob/f803e8240befeeb7b9522df0d87b79ff2149b2b0/pytrickle/health.py#L16-L178

`StreamHealthManager` is used by `TrickleStreamManager` to update the overall state. It is now used more directly by comfystream here https://github.com/livepeer/comfystream/blob/8c024bc64a70b8506049151d8ca7818e87b557f3/server/trickle_stream_manager.py#L14-L75

`active_streams` is used by `health manager` in some cleanup logic here https://github.com/livepeer/pytrickle/blob/f803e8240befeeb7b9522df0d87b79ff2149b2b0/pytrickle/health.py#L142-L147

Reply:
We can keep it and work through it when we get to looking harder at the Manager side of this.